dbgd.c

```diff
@@ -211,21 +211,55 @@ static int dbgd_mem_read(Dbg__Request *req, Dbg__Response *res)
     res->address = req->address;
     res->has_address = 1;
 
-    res->size = 1; /* FIXME: add word, dword, qword support */
-    res->has_size = 1;
+    res->data.len  = req->size;
+    res->data.data = malloc(res->data.len);
```

Comment:
Not a protobuf expert (making nxdk-rdt was my first endevour), but I think it might not `free` whatever you put here after it serializes it, so we would need to `free` it ourselves somehow. That's kinda messy right now though because its not until after this function returns that the structure is serialized in the caller, would need some rearchitecting to fix it... came across the same problem when I was trying to add the screenshot thing.